### PR TITLE
[Service Connector] `az webapp connection list`: Fix interactive input

### PR DESCRIPTION
**Pull request by wchigit**
**Related command**
az webapp/containerapp/spring/aks connection list
az aks connection create target
az source connection create storage-blob/postgres/keyvault --new

**Description**
This PR fixes the bugs of CLI interactive commands:
az source connection list/show: opt_out_list not in namespace
az aks connection create target: At least one auth info is needed
az source connection create storage-blob/postgres/keyvault --new: At least one auth info is needed

**Testing Guide**
Test the following commands interactively:
az webapp connection list
az aks connection create storage-blob
az webapp connection create storage-blob --new

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Service Connector

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py [hunk 1]
```diff
@@ -286,18 +286,18 @@ def get_local_context_value(cmd, arg):
 def opt_out_auth(namespace):
     '''Validate if config and auth are both opted out
     '''
-    opt_out_list = namespace.opt_out_list
```
Comment (xfz11) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py:289]: Can we just change the line `opt_out_list = getattr(namespace, 'opt_out_list', None)`

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py [hunk 2]
```diff
@@ -444,11 +444,12 @@ def validate_target_resource_id(cmd, namespace):
 def validate_opt_out_auth_and_config(namespace):
     '''Validate if config and auth are both opted out
     '''
-    opt_out_list = namespace.opt_out_list
-    if opt_out_list is not None and \
-            OPT_OUT_OPTION.AUTHENTICATION.value in opt_out_list and \
-            OPT_OUT_OPTION.CONFIGURATION_INFO.value in opt_out_list:
-        return True
+    if getattr(namespace, 'opt_out_list', None) is not None:
+        opt_out_list = namespace.opt_out_list
```
Comment (xfz11) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py:448]: The same as above comments

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py [hunk 3]
```diff
@@ -800,6 +821,7 @@ def apply_addon_params(cmd, namespace, arg_values):
     '''Set addon command missing args
     '''
     apply_source_args(cmd, namespace, arg_values)
+    apply_auth_args(cmd, namespace, arg_values)
```
Comment (xfz11) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py:824]: Why addon needs auth args?

**Review by xfz11 — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py (hunk 3 above)
Comment (wchigit) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py:824]: Create connection needs an auth type.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**